src/state/slices/portfolioSlice/portfolioSliceCommon.ts

```diff
@@ -35,8 +37,12 @@ export type AccountSpecifier = string
 
 export type PortfolioAccounts = {
   byId: {
-    // asset ids belonging to an account
-    [k: AccountSpecifier]: CAIP19[]
+    [k: AccountSpecifier]: {
+      // asset ids belonging to an account
+      assetIds: CAIP19[]
+      // mapping(accountSpecifier -> StakingDataByAccountSpecifier)
+      stakingDataByAccountSpecifier?: StakingDataByAccountSpecifier
```

Comment:
this should probably be `byValidatorId` - we're already inside something scoped to the accountSpecifier. when this join was on the stakingData side, this made sense, but now the join goes the other direction; to the validator id

Reply:
Will go with `stakingData` as an intermediate step to have less broken things but will keep this open to change it in a few commits